### PR TITLE
feat: support to set architecture of SDK executable

### LINKED ISSUE #146: bug: can't run flutter on macOS
The system architecture of GitHub Action is `x86` not `arm64`, so should download the x86 archive.
Error: `Bad CPU type in executable`
PR: https://github.com/subosito/flutter-action/pull/147
image
image

### PR DESCRIPTION
**Pull request by SiongSng**
Close #146

### CODE REVIEW

**Comment by subosito** [comment]
Hi @SiongSng, test checks were failed; it would be great if you could fix that.

**Comment by SiongSng** [comment]
> Hi @SiongSng, test checks were failed; it would be great if you could fix that.

OK

**Comment by SiongSng** [comment]
@subosito Please approve the workflow thanks

**Comment by SiongSng** [comment]
@subosito Please approve the workflow thanks

**Comment by SiongSng** [comment]
@subosito Please approve the workflow thanks

**Comment by subosito** [comment]
@SiongSng please revert changes for `.github/workflows/workflow.yml` so we can focus on the changes themselves. Thanks.

File: setup.sh [hunk 1]
```diff
@@ -14,29 +14,34 @@ normalize_version() {
 }
 
 latest_version() {
-  jq --arg channel "$1" '.releases | map(select(.channel==$channel)) | first'
+  jq --arg channel "$1" --arg dart_sdk_arch "$ARCHITECTURE" '.releases | map(select(.channel==$channel) | select(.dart_sdk_arch==$dart_sdk_arch)) | first'
```
Comment (subosito) [inline setup.sh:17]: I think we can use something like `select(.dart_sdk_arch as $sdk_arch | $arch | index($sdk_arch))` with argument like:

- `--argjson arch '["x64", null]'` for x64 and default value, or
- `--argjson arch '["arm64"]'` when looking for arm64

File: setup.sh [hunk 2]
```diff
@@ -14,29 +14,34 @@ normalize_version() {
...
 }
 
 wildcard_version() {
-  if [[ $1 == any ]]; then
-    jq --arg version "^$2" '.releases | map(select(.version | test($version))) | first'
+  if [[ $2 == *"v"* ]] || [[ $2 == 1 ]] || [[ $2 == 0 ]]; then  # is legacy version format
```
Comment (subosito) [inline setup.sh:21]: Using the above approach, I think we can simplify it. Having fewer conditionals is always better.

File: setup.sh [hunk 3]
```diff
@@ -14,29 +14,34 @@ normalize_version() {
...
   else
     wildcard_version $1 $2
   fi
 }
 
 get_version_manifest() {
-  releases_manifest=$(curl --silent --connect-timeout 15 --retry 5 $MANIFEST_URL)
-  version_manifest=$(echo $releases_manifest | get_version $1 $(normalize_version $2))
-
+ releases_manifest=$(curl --silent --connect-timeout 15 --retry 5 $MANIFEST_URL)
```
Comment (subosito) [inline setup.sh:43]: Please revert changes that are not related to the actual issue, git diff will be much easier.

File: setup.sh [hunk 4]
```diff
@@ -89,6 +94,7 @@ done
 
 CHANNEL="${@:$OPTIND:1}"
 VERSION="${@:$OPTIND+1:1}"
+ARCHITECTURE="${@:$OPTIND+2:1}"
```
Comment (subosito) [inline setup.sh:97]: I think a shorter form is better, like `ARCH`

**Review by subosito — CHANGES_REQUESTED** [review]
For the functionality, I think it's already OK. Now, let's make it prettier and easier to maintain 👍
